lib/schemas.js

```diff
@@ -129,22 +150,49 @@ function getSchemaAnyway (schema, jsonShorthand) {
  *
  * @param {object} context the request context
  * @param {number} statusCode the http status code
+ * @param {string} contentType the reply content type
  * @returns {function|boolean} the right JSON Schema function to serialize
  * the reply or false if it is not set
  */
-function getSchemaSerializer (context, statusCode) {
+function getSchemaSerializer (context, statusCode, contentType) {
   const responseSchemaDef = context[kSchemaResponse]
   if (!responseSchemaDef) {
     return false
   }
   if (responseSchemaDef[statusCode]) {
+    if (responseSchemaDef[statusCode].constructor === Object) {
+      const mediaName = contentType.split(';')[0]
```

Comment:
you name it mediaName here and in the other cases only type? I assume they are everywhere the same "content" so maybe rename everywhere to mediaName?

Is this function called on every call of request to fastify or only on initializiation? Why do we need to split the string with a semicolon?
If this gets called only on initialization then I am fine with this. But if we run this on every call, then I would prefer not to run split on every call. I would rather have a new function called getMediaNameByContentType which caches the split result in a Map. 

What do you guys think
@mcollina 
@climba03003 
@jsumners

Reply:
> you name it mediaName here and in the other cases only type? I assume they are everywhere the same "content" so maybe rename everywhere to mediaName?

Yes, they're all referring to the content-type name. I will commit a change renaming `type` to `media-name`.
> 
> Is this function called on every call of request to fastify or only on initializiation? Why do we need to split the string with a semicolon? If this gets called only on initialization then I am fine with this. But if we run this on every call, then I would prefer not to run split on every call. I would rather have a new function called getMediaNameByContentType which caches the split result in a Map.
> 

On every call. I needed to split to get only the `content-type`, without the charset, if present:
```JS
application/json; charset=utf-8
```
I didn't get what you mean by having a new function for that, kindly, can you elaborate on this? wouldn't we need to call that function also on every request?

> What do you guys think @mcollina @climba03003 @jsumners

Reply:
https://github.com/fastify/fastify/blob/a4bb8ad161f6de802e4617fff6b4477fd1c084b2/lib/reply.js#L177-L188

It seems like Fastify is always appending the charset, so it will always be present

Reply:
You can not be sure of that. 

If i remember correctly it works like this

1. Browser sends request with accept header containing e.q. "application/json; q=1"
2. fastify/accept-negotiator determines according to the accept header which content type it should try to return
3. We are now in the code of yours

I have to check your code and tests...

Reply:
As per what I understand from that piece of code, Fastify is checking the content-type been set from the handler whether it's present or not, then appending the charset if not present, I don't think that Fastify would try to return a content-type automatically based on the Accept header because this not always the needed the use-case I think. I might be mistaken though, I will take another look later today on the accept-negotiator considering the points you have mentioned to be sure of that